### PR TITLE
Add a new configuration item: proxy.http.port

### LINKED ISSUE #25873: [Bug]: [restful]When authorizationEnabled is not enabled, token is still needed for restful api
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:2.2.0-20230724-1f1ac8f1
- Deployment mode(standalone or cluster):
- MQ type(rocksmq, pulsar or kafka):    
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

When authorizationEnabled is not enabled, the token is still needed for restful API

### Expected Behavior

It would be better to skip auth in restful API if authorizationEnabled is false.

The user does not have authentication turned on, in this case, other sdk's connect without username and password, but restful requires the default `root:Milvus` as token. This is inconsistent with the client's behavior.

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by PowderLi**
/kind bug
fix #25873
bring by #24761

### CODE REVIEW

File: internal/distributed/proxy/service.go [hunk 1]
```diff
@@ -132,6 +133,10 @@ func authenticate(c *gin.Context) {
 			return
 		}
 	}
+	if !proxy.Params.CommonCfg.AuthorizationEnabled {
```
Comment (jaime0815) [inline internal/distributed/proxy/service.go:136]: 1. Why need to set root as a default user if authorization is disabled? 
2.  is it reasonable to set the current name to root after parsing the user name successfully if authorization is disabled

File: internal/distributed/proxy/service.go [hunk 2]
```diff
@@ -407,13 +412,67 @@ func (s *Server) init() error {
...
+				s.httpListener, listenErr = net.Listen("tcp", ":"+strconv.Itoa(HTTPParams.Port))
+				if listenErr != nil {
+					log.Error("Proxy server(http) failed to listen on", zap.Error(err), zap.Int("port", HTTPParams.Port))
+					return err
+				}
+			} else if Params.TLSMode == 1 {
+				creds, err := tls.LoadX509KeyPair(Params.ServerPemPath, Params.ServerKeyPath)
+				if err != nil {
+					log.Error("proxy can't create creds", zap.Error(err))
+					log.Error("proxy can't create creds", zap.Error(err))
```
Comment (jaime0815) [inline internal/distributed/proxy/service.go:434]: duplicated log

File: internal/distributed/proxy/httpserver/handler_v1.go [hunk 3]
```diff
@@ -619,11 +633,13 @@ func (h *Handlers) search(c *gin.Context) {
 		GuaranteeTimestamp: BoundedTimestamp,
 		Nq:                 int64(1),
 	}
-	username, _ := c.Get(ContextUsername)
-	_, authErr := proxy.PrivilegeInterceptorWithUsername(c, username.(string), &req)
-	if authErr != nil {
-		c.JSON(http.StatusUnauthorized, gin.H{HTTPReturnCode: Code(authErr), HTTPReturnMessage: authErr.Error()})
-		return
+	if proxy.Params.CommonCfg.AuthorizationEnabled {
```
Comment (jaime0815) [inline internal/distributed/proxy/httpserver/handler_v1.go:636]: we'd better add a new checkAuthorization method to reduce duplicated code

**Comment by PowderLi** [comment]
rerun ut

File: internal/distributed/proxy/httpserver/handler_v1.go [hunk 4]
```diff
@@ -47,7 +47,7 @@ func (h *Handlers) describeCollection(c *gin.Context, dbName string, collectionN
 		DbName:         dbName,
 		CollectionName: collectionName,
 	}
-	if needAuth {
+	if needAuth && proxy.Params.CommonCfg.AuthorizationEnabled {
```
Comment (jaime0815) [inline internal/distributed/proxy/httpserver/handler_v1.go:50]: Why needAuth is true if the requested prefix is /vector/collections/describe? yet other requests don't need to check auth.

File: internal/distributed/proxy/httpserver/handler_v1.go [hunk 5]
```diff
@@ -190,11 +192,13 @@ func (h *Handlers) createCollection(c *gin.Context) {
 		ShardsNum:        ShardNumDefault,
 		ConsistencyLevel: commonpb.ConsistencyLevel_Bounded,
 	}
-	username, _ := c.Get(ContextUsername)
-	_, authErr := proxy.PrivilegeInterceptorWithUsername(c, username.(string), &req)
-	if authErr != nil {
-		c.JSON(http.StatusUnauthorized, gin.H{HTTPReturnCode: Code(authErr), HTTPReturnMessage: authErr.Error()})
-		return
+	if proxy.Params.CommonCfg.AuthorizationEnabled {
```
Comment (jaime0815) [inline internal/distributed/proxy/httpserver/handler_v1.go:195]: Suggest using a method to check auth to reduce duplicated code.

**Comment by jaime0815** [comment]
To ensure that GRPC and restful pass username as the requested tenant ID through context,  we still need to process[ this method](https://github.com/milvus-io/milvus/blob/0ce6dda785ab9112ff98ddaec394e11642b9e19c/internal/proxy/util.go#L818)  to make it work well if it is a restful request.

**Comment by PowderLi** [comment]
duplicated #26008

**Pull request closed without merging**